jenkins/spark-tests.sh

```diff
@@ -211,6 +211,18 @@ run_iceberg_tests() {
   fi
 }
 
+ # Test spark-avro with documented way of deploying at run time via --packages option from Maven
+run_avro_tests() {
+  unset TEST_PARALLEL  # Enable auto spark local parallel in run_pyspark_from_build.sh
```

Comment:
`SPARK_SUBMIT_FLAGS` and TEST_PARALLEL > 0 are not compatible, see the thread on #6403

Reply:
refined

Reply:
we do not require it must be parallel in this PR as we are testing `--package avro`. 

For this place to let run run_pyspark_from_build to set the parallelism may not always be a good idea as it try to set as many as parallelism as possible, the actually it would decrease the performance if over some specific value and could potentially caused host machine memory OOM on different GPU types. I guess we could hardcode it to 4 or 5 here

We can discuss about how to better parallelize tests like iceberg and avro in #6403. But for this PR lets keep it simple~

Reply:
agreed. we need to drop `unset TEST_PARALLEL`  in this PR

Reply:
removed